hive-metastore/src/test/java/org/apache/iceberg/hive/TestHiveCommitLocks.java

```diff
@@ -0,0 +1,186 @@
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.when;
+
+public class TestHiveCommitLocks extends HiveTableBaseTest {
+  HiveTableOperations ops = null;
+  HiveTableOperations spyOps = null;
+  HiveClientPool spyClientPool = null;
+  HiveMetaStoreClient spyClient = null;
+  TableMetadata metadataV1 = null;
+  TableMetadata metadataV2 = null;
+
+  long dummyLockId = 500L;
+  LockResponse waitLockResponse = new LockResponse(dummyLockId, LockState.WAITING);
+  LockResponse acquiredLockResponse = new LockResponse(dummyLockId, LockState.ACQUIRED);
+  LockResponse notAcquiredLockResponse = new LockResponse(dummyLockId, LockState.NOT_ACQUIRED);
+
+  @Before
+  public void before() throws Exception {
+    Table table = catalog.loadTable(TABLE_IDENTIFIER);
+    ops = (HiveTableOperations) ((HasTableOperations) table).operations();
+    String dbName = TABLE_IDENTIFIER.namespace().level(0);
+    String tableName = TABLE_IDENTIFIER.name();
+    Configuration overriddenHiveConf = new Configuration(hiveConf);
+    overriddenHiveConf.setLong("iceberg.hive.lock-timeout-ms", 10 * 1000);
+    overriddenHiveConf.setLong("iceberg.hive.lock-check-min-wait-ms", 50);
+    overriddenHiveConf.setLong("iceberg.hive.lock-check-max-wait-ms", 5 * 1000);
+    overriddenHiveConf.setDouble("iceberg.hive.lock-check-backoff-scale-factor", 3.0);
+
+    metadataV1 = ops.current();
+
+    table.updateSchema()
+        .addColumn("n", Types.IntegerType.get())
+        .commit();
+
+    ops.refresh();
+
+    metadataV2 = ops.current();
+
+    Assert.assertEquals(2, ops.current().schema().columns().size());
+
+    spyClientPool = spy(new HiveClientPool(1, overriddenHiveConf));
+    AtomicReference<HiveMetaStoreClient> spyClientRef = new AtomicReference<>();
+
+    when(spyClientPool.newClient()).thenAnswer(invocation -> {
+      HiveMetaStoreClient client = (HiveMetaStoreClient) invocation.callRealMethod();
+      spyClientRef.set(spy(client));
+      return spyClientRef.get();
+    });
+
+    spyOps = spy(new HiveTableOperations(overriddenHiveConf, spyClientPool, ops.io(), catalog.name(),
+        dbName, tableName));
+    spyClientPool.run(client -> client.isLocalMetaStore()); // To ensure new client is created.
+    Assert.assertNotNull(spyClientRef.get());
+
+    spyClient = spyClientRef.get();
+  }
+
+  @After
+  public void cleanup() {
+    try {
+      spyClientPool.close();
+    } catch (Throwable t) {
+      // Ignore any exception
+    }
+  }
+
+  @Test
+  public void testLockAcquisitionAtFirstTime() throws TException, InterruptedException {
+    doReturn(acquiredLockResponse).when(spyClient).lock(any());
+    doNothing().when(spyOps).doUnlock(eq(dummyLockId));
+
+    spyOps.doCommit(metadataV2, metadataV1);
+
+    Assert.assertEquals(1, spyOps.current().schema().columns().size()); // should be 1 again
+  }
+
+  @Test
+  public void testLockAcquisitionAfterRetries() throws TException, InterruptedException {
```

Comment:
Is `InterruptedException` needed?

Reply:
yes, here the [HiveTableOperations.doUnlock](https://github.com/apache/iceberg/blob/master/hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java#L340) method throws `InterruptedException`.